native/src/utils/renderHtml.ts

```diff
@@ -71,34 +84,143 @@ const renderJS = (cacheDictionary: Record<string, string>) => `
-      window.ReactNativeWebView.postMessage(JSON.stringify({ type: '${HEIGHT_MESSAGE_TYPE}', height: height }));
-      container.setAttribute('style', '');
+      const height = container.getBoundingClientRect().height - 2
+      window.ReactNativeWebView.postMessage(JSON.stringify({ type: '${HEIGHT_MESSAGE_TYPE}', height: height }))
+      container.setAttribute('style', '')
     }
 
-    window.addEventListener('load', adjustHeight);
-    window.addEventListener('resize', adjustHeight);
-    var details = document.querySelectorAll("details")
-    details.forEach(detail => detail.addEventListener("toggle", adjustHeight))
+    window.addEventListener('load', adjustHeight)
+    window.addEventListener('resize', adjustHeight)
+    const details = document.querySelectorAll('details')
+    details.forEach(detail => detail.addEventListener('toggle', adjustHeight))
   })();
+
+  (function handleIframes() {
+    function capitalizeFirstLetter(word) {
+      return word.charAt(0).toUpperCase() + word.slice(1)
+    }
+
+    function showMessage(text, element, className) {
+      const textNode = document.createTextNode(text)
+      if (className) {
+        element.classList.add(className)
+      }
+      element.appendChild(textNode)
+    }
+
+    function showSettingsButton(element) {
+      function onClickHandler () {
+        window.ReactNativeWebView.postMessage(JSON.stringify({ type: '${SETTINGS_MESSAGE_TYPE}', openSettings: true }))
+      }
+
+      const buttonLabel = ${JSON.stringify(t('layout:settings'))}
+      const button = document.createElement('button')
+      button.name = 'opt-in-settings-button'
+      button.innerHTML = buttonLabel
+      button.id = 'opt-in-settings-button'
+      button.onclick = onClickHandler
+      element.appendChild(button)
+    }
+
+    function showMessageWithSettings(text, iframeContainer) {
+      showMessage(text, iframeContainer, 'iframe-info-text')
+      showSettingsButton(iframeContainer)
+    }
+
+    function showOptIn(text, iframeContainer, source) {
+      function onClickHandler () {
+        window.ReactNativeWebView.postMessage(JSON.stringify({ type: '${IFRAME_MESSAGE_TYPE}', allowedSource: { type: source, allowed: true } }))
+      }
+
+      const checkbox = document.createElement('input')
+      checkbox.type = 'checkbox'
+      checkbox.name = 'opt-in-checkbox'
+      checkbox.id = 'opt-in-checkbox'
+      checkbox.onclick = onClickHandler
+      const label = document.createElement('label')
+      label.htmlFor = 'opt-in-checkbox'
+      label.appendChild(document.createTextNode(text))
+      iframeContainer.appendChild(label)
+      iframeContainer.appendChild(checkbox)
+    }
+
+    function showBlockMessageWithSettings(text, iframeContainer) {
+      showMessage(text, iframeContainer)
+      showSettingsButton(iframeContainer)
+    }
+
+    function handleAllowedIframeSources(iframe, allowedExternalSourcePermissions, iframeSource, iframeContainer) {
```

Comment:
Maybe a better name is actually `supportedIframeSources` instead of allowed since they are only allowed from our perspective and not from the users perspective and this might make the logic more clear. At least I was confused about the two different types of "allowed". If you agree, please rename the params and build config properties as well.

Reply:
alright i also changed them in build config